k8s/upgrade/src/bin/upgrade-job/helm/yaml/yq.rs

```diff
@@ -0,0 +1,172 @@
+use crate::{
+    common::error::{
+        NotYqV4, RegexCompile, Result, U8VectorToString,
+        YqCommandExec, YqMergeCommand, YqMergeOutputSuffixRemoval, YqVersionCommand, NotAValidYamlKeyForStringValid, YqSetCommand,
+    },
+    vec_to_strings,
+};
+use regex::Regex;
+use snafu::{ensure, ResultExt};
+use std::{path::Path, process::Command, str};
+use std::ops::Deref;
+use tracing::debug;
+
+pub(crate) struct YamlKey (String);
+
+impl TryFrom<String> for YamlKey {
+    type Error = crate::common::error::Error;
+
+    fn try_from(value: String) -> Result<Self> {
+        let yaml_key_regex = r"^(\.[a-zA-Z0-9])+$";
+        ensure!(Regex::new(yaml_key_regex)
+            .context(RegexCompile {
+                expression: yaml_key_regex.to_string(),
+            })?
+            .is_match(value.as_str()), NotAValidYamlKeyForStringValid { key: value.clone() });
+        Ok(YamlKey(value))
+    }
+}
+
+impl AsRef<YamlKey> for YamlKey {
+    fn as_ref(&self) -> &YamlKey {
+        self.deref()
+    }
+}
+
+impl YamlKey {
+    fn literal_value(&self) -> String {
```

Comment:
a bit confusing, isn't literal in rust an "string"?
I suggest using inner, or simply ToString or perhaps using deref trait allowing you to borrow &str

Reply:
Good one, the Deref trait seems to be more intuitive for this one.